### DISCUSSION TITLE
Browser Subprocess Crashing on Startup

### DISCUSSION THREAD

**Discussion by cplotts** [cefsharp/CefSharp#4448] [General]
We are seeing the CefSharp.BrowserSubprocess.exe crashing when we initialize CefSharp.

This is only happening on a few machines ... but the problem seems to be spreading. It is almost like our IT department is rolling out an update or something that is conflicting with CefSharp.

It is not happening on any development machine at my disposal ... otherwise I would debug it.

Does anyone have any ideas or suggestions on how to troubleshoot this? Or any idea on what may be happening?

We are using 94.4.50 of CefSharp.Wpf.

Thank you!
Cory (https://github.com/cplotts)

![image](https://user-images.githubusercontent.com/1655985/229579517-e64b147f-626c-49ee-84a0-5733885bcb99.png)

![image](https://user-images.githubusercontent.com/1655985/229579544-7c06d1de-8b0e-4d5c-a976-eb27b900a4c6.png)

![image](https://user-images.githubusercontent.com/1655985/229579567-f9c24b11-74ef-43af-862b-9017bb6bae79.png)

**Comment by cplotts** [comment]
I see that there is a whole section on https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting the crashes in CefSharp.BrowserSubprocess.exe ... so I will see what additional information I can get.

**Comment by cplotts** [comment]
![image](https://user-images.githubusercontent.com/1655985/229601073-00ae7fa9-dbd3-413a-9c9a-f8041cc78897.png)

**Reply by cplotts** [reply]
Looks like I'm getting a Win32Exception when trying to run the CefSharp.BrowserSubprocess.

Application: CefSharp.BrowserSubprocess.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ComponentModel.Win32Exception
   at System.Diagnostics.ProcessManager.OpenProcess(Int32, Int32, Boolean)
   at System.Diagnostics.Process.GetProcessHandle(Int32, Boolean)
   at System.Diagnostics.Process.OpenProcessHandle(Int32)
   at System.Diagnostics.Process.get_Handle()
   at CefSharp.BrowserSubprocess.SubProcess.Run()
   at CefSharp.BrowserSubprocess.BrowserSubprocessExecutable.Main(System.Collections.Generic.IEnumerable`1<System.String>, CefSharp.RenderProcess.IRenderProcessHandler)
   at CefSharp.BrowserSubprocess.Program.Main(System.String[])

**Reply by amaitland** [reply]
> This is only happening on a few machines ... but the problem seems to be spreading. It is almost like our IT department is rolling out an update or something that is conflicting with CefSharp.

It's something to consider. Antivirus update maybe? Some new security restrictions?

> Application: CefSharp.BrowserSubprocess.exe
> Framework Version: v4.0.30319
> Description: The process was terminated due to an unhandled exception.
> Exception Info: System.ComponentModel.Win32Exception
>    at System.Diagnostics.ProcessManager.OpenProcess(Int32, Int32, Boolean)
>    at System.Diagnostics.Process.GetProcessHandle(Int32, Boolean)
>    at System.Diagnostics.Process.OpenProcessHandle(Int32)
>    at System.Diagnostics.Process.get_Handle()
>    at CefSharp.BrowserSubprocess.SubProcess.Run()
>    at CefSharp.BrowserSubprocess.BrowserSubprocessExecutable.Main(System.Collections.Generic.IEnumerable`1<System.String>, CefSharp.RenderProcess.IRenderProcessHandler)
>    at CefSharp.BrowserSubprocess.Program.Main(System.String[])

Looks like the `auto hInstance = Process::GetCurrentProcess()->Handle;` call is failing for some reason. Security issue maybe?

Basically it's just a call to [Process.Handle](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.handle?view=net-6.0)

**Reply by cplotts** [reply]
Yes, I'm engaging our IT people on this issue ... I strongly suspect our security polices and/or software are at work here.

I thought I would post a discussion here ... just in case.
